src/hotspot/share/opto/parse1.cpp

```diff
@@ -2174,16 +2173,50 @@ PhiNode *Parse::ensure_memory_phi(int idx, bool nocreate) {
   return phi;
 }
 
+// Passive Materialization
+// ------------------------
+// C2 has to materialize a virtual object at the merging point because the object has been materialized in
+// any other predecessor. For instance:
+//
+// obj(virtual)     obj(materialized)
+//              \      /
+// Reion(,c1,c2) |    |
+//      \        |    |
+//       \c      |    |
+//   o3 = Phi   (o1, o2)
+//
+// Node* var is o1;
+// r[pnum] is c1;
+//
+// There are 2 cases:
+// 1) materialize m from map().
```

Comment:
Use of `m` and `n` are confusing because those variables are in merge_common(), but not this function